tests/Doctrine/Tests/ORM/CommitOrderCalculatorTest.php

```diff
@@ -100,6 +100,33 @@ public function testCommitOrdering3()
         // We want to perform a strict comparison of the array
         $this->assertContains($sorted, $correctOrders, '', false, true, true);
     }
+    
+    public function testCommitOrdering4()
```

Comment:
I can't say I really understand this PR, but here is some help for those who can:

```
       +---------+
       |         |
    +--+ Class 4 <---+
    |  |         |   |
  1 |  +---------+   | 0
    |                |
    v                |
+---+-----+    +-----+---+
|         |    |         |
| Class 1 +--->+ Class 2 |
|         | 0  |         |
+---+-----+    +-----+---+
    ^                |
    |                |
  0 |                | 0
    |  +---------+   |
    |  |         |   |
    +--+ Class 3 +<--+
       |         |
       +---------+

```

Maybe embed this as a comment if you find it useful?

Reply:
Thank you for the suggestion. I embedded this graph in a comment within the test.